### ISSUE TITLE
Saving strapi item adds it to index without any checks

### ISSUE THREAD

**Issue by FREEZX** [meilisearch/strapi-plugin-meilisearch#264]
**Description**
Saving any strapi item adds it to the index without checking it's published state, if it's the right locale, etc. So running "Update" in the meilisearch plugin gives different results than using strapi normally.

**Expected behavior**
When I save a draft item, it should not go in the index.
When I save an item in a locale which is not indexed by refreshing the indexes, it should not go in the index also.

**Current behavior**
When saving a draft item, it goes in the index.
When I save a localized page, it gets added to the index. (See issue https://github.com/meilisearch/strapi-plugin-meilisearch/issues/263)

**Environment (please complete the following information):**
- OS: Ubuntu 20.04 LTS
- MeiliSearch version: 0.20.0
- strapi-plugin-meilisearch version: 0.3.3
- Strapi version: [e.g. v3.6.5]

**Comment by bidoubiwa** [comment]
@FREEZX thanks for this issue as well! 

You are right, we only check for the publish state when indexing the whole collection. 
What are the other parameters we should check before indexing the row to MeiliSearch?

**Comment by paulpopus** [comment]
> What are the other parameters we should check before indexing the row to MeiliSearch?

Could that not be a custom setting? I'm thinking ahead of things like editorial workflows where content might go in various stages before "publicly published" if that makes sense.

**Comment by bidoubiwa** [comment]
It could be an option. We can give the possibility to the user to add a wrapper function around the function that adds the documents to MeiliSearch. This should be a bit further discussed on the best way we can achieve that with strapi's API and also if it will not made easier with strapi 4 as we will eventually focus our attention more on strapi 4.

**Comment by paulpopus** [comment]
That sounds like a good approach.

> also if it will not made easier with strapi 4 as we will eventually focus our attention more on strapi 4

Depending on when Strapi 4 releases, I think the minimum expectation would be that Drafts don't get indexed even for V3.

**Comment by paulpopus** [comment]
In the meantime if someone needs a quick fix for production, I've attached a patch below. You can use the [patch-package](https://www.npmjs.com/package/patch-package) node module to apply this patch. Copy this code into `patches/strapi-plugin-meilisearch+0.3.3.patch`

```
diff --git a/node_modules/strapi-plugin-meilisearch/services/lifecycles.js b/node_modules/strapi-plugin-meilisearch/services/lifecycles.js
index e2bf53e..ae01950 100644
--- a/node_modules/strapi-plugin-meilisearch/services/lifecycles.js
+++ b/node_modules/strapi-plugin-meilisearch/services/lifecycles.js
@@ -5,13 +5,15 @@
  */
 
 async function afterCreate(result, collection, httpClient) {
-  try {
-    await httpClient.addDocuments({
-      indexUid: collection,
-      data: [result],
-    })
-  } catch (e) {
-    console.error(e)
+  if(result.published_at) {
+    try {
+      await httpClient.addDocuments({
+        indexUid: collection,
+        data: [result],
+      })
+    } catch (e) {
+      console.error(e)
+    }
   }
 }
 
@@ -31,13 +33,15 @@ async function afterDelete(result, collection, httpClient) {
 }
 
 async function afterUpdate(result, collection, httpClient) {
-  try {
-    await httpClient.addDocuments({
-      indexUid: collection,
-      data: [result],
-    })
-  } catch (e) {
-    console.error(e)
+  if(result.published_at) {
+    try {
+      await httpClient.addDocuments({
+        indexUid: collection,
+        data: [result],
+      })
+    } catch (e) {
+      console.error(e)
+    }
   }
 }
```

**Comment by bidoubiwa** [comment]
fixed by: #288
